controller/src/main/java/org/jboss/as/controller/PathElement.java

```diff
@@ -177,6 +177,15 @@ public boolean matches(Property property) {
     }
 
     /**
+     * Determine whether the given element matches this element.
```

Comment:
This needs javadoc to describe the matching algorithm, i.e. that it differs from equals in that wildcards are involved.

A couple subtleties worth noting:

1) This only differs from equals(pe) if this.isWildCard() but not if only pe.isWildCard().
2) PathElement also has the isMultiTarget() concept, which is goes beyond isWildCard() and isn't factored into this matches() method. Perhaps it should be though, since this is a general purpose public method in a very widely used class.

The way that isMultiTarget() goes beyond isWildCard() isn't relevant to the CapabilityRegistry use case here, as no such addresses will be stored in the registry.